github/messages_test.go

```diff
@@ -79,3 +81,110 @@ func TestValidatePayload(t *testing.T) {
 		}
 	}
 }
+
+func TestGetWebhookEvent(t *testing.T) {
+	tests := []struct {
+		payload     interface{}
+		messageType string
+	}{
+		{
+			payload:     &CommitCommentEvent{},
+			messageType: "commit_comment",
+		},
+		{
+			payload:     &CreateEvent{},
+			messageType: "create",
+		},
+		{
+			payload:     &DeleteEvent{},
+			messageType: "delete",
+		},
+		{
+			payload:     &DeploymentEvent{},
+			messageType: "deployment",
+		},
+
+		{
+			payload:     &DeploymentStatusEvent{},
+			messageType: "deployment_status",
+		},
+		{
+			payload:     &ForkEvent{},
+			messageType: "fork",
+		},
+		{
+			payload:     &GollumEvent{},
+			messageType: "gollum",
+		},
+		{
+			payload:     &IssueCommentEvent{},
+			messageType: "issue_comment",
+		},
+		{
+			payload:     &IssuesEvent{},
+			messageType: "issues",
+		},
+		{
+			payload:     &MemberEvent{},
+			messageType: "member",
+		},
+		{
+			payload:     &MembershipEvent{},
+			messageType: "membership",
+		},
+		{
+			payload:     &PageBuildEvent{},
+			messageType: "page_build",
+		},
+		{
+			payload:     &PublicEvent{},
+			messageType: "public",
+		},
+		{
+			payload:     &PullRequestEvent{},
+			messageType: "pull_request",
+		},
+		{
+			payload:     &PullRequestReviewCommentEvent{},
+			messageType: "pull_request_review_comment",
+		},
+		{
+			payload:     &PushEvent{},
+			messageType: "push",
+		},
+		{
+			payload:     &ReleaseEvent{},
+			messageType: "release",
+		},
+		{
+			payload:     &RepositoryEvent{},
+			messageType: "repository",
+		},
+		{
+			payload:     &StatusEvent{},
+			messageType: "status",
+		},
+		{
+			payload:     &TeamAddEvent{},
+			messageType: "team_add",
+		},
+		{
+			payload:     &WatchEvent{},
+			messageType: "watch",
+		},
+	}
+
+	for _, test := range tests {
+		p, err := json.Marshal(test.payload)
+		if err != nil {
+			t.Fatalf("Marshal(%#v): %v", test.payload, err)
```

Comment:
For all of these test outputs, you might prefer "%+v" over "%#v" since we are dealing with pointers to structs... but your call... see which you output you think is clearer.